subprojects/test-runner/client/src/test/kotlin/com/avito/runner/finalizer/FinalizerImplTest.kt

```diff
@@ -0,0 +1,91 @@
+package com.avito.runner.finalizer
+
+import com.avito.report.NoOpReportLinkGenerator
+import com.avito.report.model.AndroidTest
+import com.avito.report.model.Incident
+import com.avito.report.model.TestRuntimeDataPackage
+import com.avito.report.model.TestStaticDataPackage
+import com.avito.report.model.createStubInstance
+import com.avito.runner.finalizer.action.WriteTaskVerdictAction
+import com.avito.runner.scheduler.runner.model.TestSchedulerResult
+import com.avito.truth.isInstanceOf
+import com.google.common.truth.Truth.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+
+internal class FinalizerImplTest {
+
+    @Test
+    fun `finalized - ok - single completed test`() {
+        val finalizerImpl = FinalizerImpl.createStubInstance()
+
+        val test = TestStaticDataPackage.createStubInstance()
+
+        val testSchedulerResult = TestSchedulerResult(
+            testsToRun = listOf(test),
+            testResults = listOf(AndroidTest.Completed.createStubInstance(testStaticData = test))
+        )
+
+        val result = finalizerImpl.finalize(testSchedulerResult)
+
+        assertThat(result).isInstanceOf<Finalizer.Result.Ok>()
+    }
+
+    @Test
+    fun `finalized - lost - single test to run, no results`(@TempDir output: File) {
+        val verdictFile = File(output, "verdict")
+
+        val finalizerImpl = createFinalizer(verdictFile)
+
+        val test = TestStaticDataPackage.createStubInstance()
+
+        val testSchedulerResult = TestSchedulerResult(
+            testsToRun = listOf(test),
+            testResults = emptyList()
+        )
+
+        val result = finalizerImpl.finalize(testSchedulerResult)
+
+        assertThat(result).isInstanceOf<Finalizer.Result.Failure>()
+        assertThat(verdictFile.readText()).contains("${test.name} ${test.device} NOT REPORTED")
+    }
+
+    @Test
+    fun `finalized - ok - single test failure`(@TempDir output: File) {
```

Comment:
```suggestion
    fun `finalized - ok - single test with incident`(@TempDir output: File) {
```
?
Not sure, if "test failure" describes the case.